src/dags/basiskaart.py

```diff
@@ -0,0 +1,235 @@
+from sql.basiskaart import SELECT_WATERDEEL_VLAK_SQL as SELECT_WATERDEEL_VLAK_SQL
+from sql.basiskaart import SELECT_WEGDEEL_VLAK_SQL as SELECT_WEGDEEL_VLAK_SQL
+from sql.basiskaart import SELECT_WEGDEEL_LIJN_SQL as SELECT_WEGDEEL_LIJN_SQL
+from sql.basiskaart import SELECT_LABELS_SQL as SELECT_LABELS_SQL
+
+
+dag_id = "basiskaart"
+owner = "dataservices"
+variables = Variable.get(dag_id, deserialize_json=True)
+tables_to_create = variables["tables_to_create"]
+source_connection="AIRFLOW_CONN_POSTGRES_BASISKAART"
+import_step = 10000
+
+
+def create_tables_from_basiskaartdb_to_masterdb(source_connection, source_select_statement, target_base_table, *args, **kwargs):
+    """
+    Source_connection contains the environment variable name (as defined in the docker-compose.yml) of the source connection i.e. AIRFLOW_CONN_POSTGRES_BASISKAART
+    Source_select_statement contains the SQL select query that will be executed on the source DB.
+    Target_base_table contains the table in master DB where the data (the result of source_select_statement execution) is inserted into.
+    """
+
+    try:
+        # setup the DB source connection
+        source_engine = create_engine(
+            env(source_connection).split("?")[0]
+        )
+    except SQLAlchemyError as e:
+        raise Exception(str(e)) from e
+
+    # fetch data from source DB
+    with source_engine.connect() as source_connection:
+        count = 0
+        cursor = source_connection.execute(source_select_statement)     
+        while True:
+            fetch_iterator = cursor.fetchmany(size=import_step)
+            batch_count = copy_data_in_batch(target_base_table, fetch_iterator)
+            count += batch_count
+            if batch_count < import_step:
+                break
+        print(f"Imported: {count}")
+
+
+def copy_data_in_batch(target_base_table, fetch_iterator):
+    
+    items = []
+    
+    # setup SQL insert values the be executed
+    for row in fetch_iterator:
+
+        row_insert = []        
+        for column_value in row:
+            
+            # escape single quote in labels, conflicts with sql insert statement i.e. Jachthaven 't Eiland
+            if 'labels' in target_base_table:                
+                p = re.compile("([\w\s]{0,})(')([\w\s]+)")
+                m = p.match(str(column_value))
+                if m:
+                    column_value = m.group(1)+'\''+m.group(2)+m.group(3)
+            
+            row_insert.append(f"'{column_value}'" if column_value else "NULL")
+        
+        items.append ('(' + ','.join(row_insert) + ')')
+            
+    result = len(items)
+
+    # execute SQL insert statement
+    if result:
+        items_sql = ",".join(items)
+        insert_sql = (
+            f"INSERT INTO {dag_id}_{target_base_table}_TEMP "
+            f"VALUES {items_sql};"
+        )
+
+        masterdb_hook = PostgresHook()
+        
+        try:
+            masterdb_hook.run(insert_sql)
```

Comment:
de PostgresHook.run() heeft een parameters 'parameters'. Die voorkomt sql injections (nu zul je daar hier niet zoveel last van hebben). Verder hoef je dan wschl. ook het escapen van de `items_sql` niet te doen. Werkt als volgt:

    run("INSERT INTO VALUES (%s, %s, %s ...)", parameters=lijst_of_params)

Het aantal `%s` is wel dynamisch, maar eenvoudig te maken met `", ".join(["%s"] * aantal)`.

Reply:
De PostgresHook.run() met parameters werkt erg goed. Ik hoef geen escapes meer uit te voeren. 

Alleen wat ik niet aan de praat krijg is in combinatie met een batch run. Het lijkt er op dat met PostgresHook.run(, parameters=lijst_of_params) met replacements alleen werkt als je deze direct uitvoerd, rij voor rij. 

De verwerking wordt daarmee wel een factor 5 a 10 trager